### PR TITLE
[WPB-662] servantify brig provider bot api

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/WPB-662

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/types-common/src/Data/Id.hs [hunk 1]
```diff
@@ -354,10 +354,11 @@ newtype BotId = BotId
       FromHttpApiData,
       Hashable,
       NFData,
-      FromJSON,
-      ToJSON,
-      Generic
+      Generic,
+      ToParamSchema
```
Comment (elland) [inline libs/types-common/src/Data/Id.hs:358]: 👍

File: libs/wire-api/src/Wire/API/Routes/Public.hs [hunk 2]
```diff
@@ -85,9 +86,19 @@ data ZType
   | ZAuthBot
   | ZAuthConv
   | ZAuthProvider
+  | -- | (Typically short-lived) access token.
+    ZAuthAccess
+
+class HasTokenType (ztype :: ZType) where
```
Comment (elland) [inline libs/wire-api/src/Wire/API/Routes/Public.hs:92]: 👍

File: changelog.d/5-internal/pr-3540 [hunk 3]
```diff
@@ -0,0 +1 @@
+The bot API is no migrated to servant
```
Comment (elland) [inline changelog.d/5-internal/pr-3540:1]:
```suggestion
The bot API is now migrated to servant
```

**Review by elland — APPROVED** [review]

File: changelog.d/5-internal/pr-3540 (hunk 3 above)
Comment (battermann) [reply changelog.d/5-internal/pr-3540]: good catch!

**Pull request merged**